lib/ansible/parsing/dataloader.py

```diff
@@ -233,35 +236,33 @@ def path_dwim(self, given):
         given = unquote(given)
         given = to_text(given, errors='surrogate_or_strict')
 
-        if given.startswith(u"/"):
-            return os.path.abspath(given)
-        elif given.startswith(u"~"):
-            return os.path.abspath(os.path.expanduser(given))
+        if RE_ABS_PATH.search(given):
```

Comment:
Don't use the regex here.  Several problems with it:
* regexes are slow.  Anytime you can replace a regex with a simple substring match use the substring functions.
* the functions above take care to use text strings.  The regex uses native strings.  That means on python2 you could have issues if nonascii characters creep in.